sym/provider/log_destination_resource.go

```diff
@@ -36,7 +36,7 @@ func LogDestinationSchema() map[string]*schema.Schema {
 
 func validateLogDestination(diags diag.Diagnostics, ld *client.LogDestination) diag.Diagnostics {
 	if ld.IntegrationId == "" {
-		if ld.Type == "http" {
+		if (ld.Type == "http" || ld.Type == "datadog") {
 			ld.IntegrationId = NullPlaceholder
```

Comment:
Hmmmm since we're changing this code... it looks like `LogDestinationSchema` is defined with `integration_id` as `Optional`. Can we change remove this line? the `NullPlaceholder` is a stupid string `<null>`, and we really should just send `integration_id = ""` or omit `integration_id` entirely. Do you mind testing to see what happens if we change this to `ld.IntegrationId = null` (is that proper Go?)


cc @Arianna2028

Reply:
Ughhhhhh 

1. To send `null`, we have to change the `LogDestination` struct to have `integration_id` be a pointer to a string, which cascades to a lot of work / we don't know Go well enough to do it quickly
2. To send an empty string, we need to change the LogDestinationSerializer to `allow_blank = True`, which is also unfortunate.

So, we shall continue eating this debt

Sad™ 😢